src/XAD/JITBackendInterface.hpp

```diff
@@ -0,0 +1,146 @@
+ *   automatic differentiation.
+ *
+ *   Copyright (C) 2010-2025 Xcelerit Computing Ltd.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU Affero General Public License as published
+ *   by the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU Affero General Public License for more details.
+ *
+ *   You should have received a copy of the GNU Affero General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ ******************************************************************************/
+
+#pragma once
+
+#include <XAD/Config.hpp>
+
+#ifdef XAD_ENABLE_JIT
+
+#include <XAD/JITGraph.hpp>
+#include <cstddef>
+
+namespace xad
+{
+
+/**
+ * @brief Abstract interface for JIT compilation backends.
+ *
+ * This interface allows plugging in different execution backends for the
+ * JIT-compiled computation graphs. Backends can range from simple interpreters
+ * to native code generators or GPU executors.
+ *
+ * ## Implementing a Custom Backend
+ *
+ * To create a custom backend, inherit from IJITBackend and implement all
+ * pure virtual methods:
+ *
+ * 1. **compile()**: Called once after graph recording is complete. Use this to:
+ *    - Translate JITGraph opcodes to your target representation
+ *    - Perform optimizations (CSE, constant folding, etc.)
+ *    - Generate native code or prepare execution structures
+ *
+ * 2. **forward()**: Execute the forward pass only. Read input values, evaluate
+ *    the graph, and write output values. Called for value-only evaluation.
+ *
+ * 3. **forwardAndBackward()**: Execute both forward and backward (adjoint) passes.
+ *    This combined method allows backends to optimize by:
+ *    - Reusing intermediate values from forward pass in backward pass
+ *    - Fusing operations across both passes
+ *    - Avoiding redundant memory allocations
+ *
+ * 4. **reset()**: Clear any compiled state. Called when the graph is cleared
+ *    or recompiled.
+ *
+ * ## Example Usage
+ *
+ * @code
+ * class MyBackend : public xad::IJITBackend {
+ * public:
+ *     void compile(const JITGraph& graph) override { ... }
+ *     void forward(...) override { ... }
+ *     void forwardAndBackward(...) override { ... }
+ *     void reset() override { ... }
+ * };
+ *
+ * // Use with JITCompiler:
+ * auto backend = std::unique_ptr<IJITBackend>(new MyBackend());
+ * xad::JITCompiler<double> jit(std::move(backend));
+ * @endcode
+ *
+ * ## Reference Implementation
+ *
+ * See JITGraphInterpreter for a reference implementation that interprets
+ * the graph directly in C++ without code generation.
+ */
+class IJITBackend
```

Comment:
For naming conventions, could you just drop the `I`? We just prefer interfaces without the leading I.